### PR TITLE
Fix panic when optimizer returns empty tree

### LINKED ISSUE #283: sql/planner: panic if optimizer returns empty tree
```
CREATE TABLE t; SELECT * FROM t WHERE [] IN [];
```

<details>
<summary>Stack trace</summary>

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x146bac9]

goroutine 1 [running]:
github.com/genjidb/genji/sql/planner.(*Tree).execute(0xc000090490, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/genjidb/genji@v0.9.0/sql/planner/tree.go:79 +0x59
github.com/genjidb/genji/sql/planner.(*Tree).Run(0xc000090470, 0x1716360, 0xc00012e000, 0xc0000c8400, 0x1b25dc8, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	github.com/genjidb/genji@v0.9.0/sql/planner/tree.go:72 +0x114
github.com/genjidb/genji/sql/query.Query.Run(0xc000090480, 0x1, 0x1, 0x0, 0x0, 0x1716360, 0xc00012e000, 0xc000078040, 0x1b25dc8, 0x0, ...)
	github.com/genjidb/genji@v0.9.0/sql/query/query.go:64 +0x24c
github.com/genjidb/genji.(*DB).Query(0xc00000e030, 0x1716360, 0xc00012e000, 0xc0000be420, 0x1f, 0x0, 0x0, 0x0, 0x1f, 0xc0001c78e8, ...)
	github.com/genjidb/genji@v0.9.0/db.go:80 +0x15b
github.com/genjidb/genji/cmd/genji/shell.(*Shell).runQuery(0xc0001d5500, 0xc0000be420, 0x1f, 0x0, 0x0)
	github.com/genjidb/genji/cmd/genji@/shell/shell.go:329 +0xbc
github.com/genjidb/genji/cmd/genji/shell.(*Shell).executeInput(0xc0001d5500, 0xc0000be420, 0x1f, 0x1, 0x2)
	github.com/genjidb/genji/cmd/genji@/shell/shell.go:266 +0x25c
github.com/genjidb/genji/cmd/genji/shell.(*Shell).execute(0xc0001d5500, 0xc0000be420, 0x1f)
	github.com/genjidb/genji/cmd/genji@/shell/shell.go:246 +0x82
github.com/c-bata/go-prompt.(*Prompt).Run(0xc00015a2d0)
	github.com/c-bata/go-prompt@v0.2.5/prompt.go:84 +0x845
github.com/genjidb/genji/cmd/genji/shell.Run(0xc000127760, 0xc000127760, 0x0)
	github.com/genjidb/genji/cmd/genji@/shell/shell.go:156 +0x586
main.main.func3(0xc0001368c0, 0xc000127700, 0x4)
	github.com/genjidb/genji/cmd/genji@/main.go:140 +0x195
github.com/urfave/cli/v2.(*App).RunContext(0xc000102480, 0x1716360, 0xc00012e000, 0xc00010e2a0, 0x1, 0x1, 0x0, 0x0)
	github.com/urfave/cli/v2@v2.2.0/app.go:315 +0x70b
github.com/urfave/cli/v2.(*App).Run(...)
	github.com/urfave/cli/v2@v2.2.0/app.go:215
main.main()
	github.com/genjidb/genji/cmd/genji@/main.go:146 +0x5f6
exit status 2
```
</details>

### PR DESCRIPTION
**Pull request by goku321**
Fixes #283

### CODE REVIEW

File: sql/query/select_test.go [hunk 1]
```diff
@@ -89,6 +89,7 @@ func TestSelectStmt(t *testing.T) {
 		{"With two non existing idents, =", "SELECT * FROM test WHERE z = y", false, `[]`, nil},
 		{"With two non existing idents, >", "SELECT * FROM test WHERE z > y", false, `[]`, nil},
 		{"With two non existing idents, !=", "SELECT * FROM test WHERE z != y", false, `[]`, nil},
+		{"With empty WHERE and IN", "SELECT * FROM test WHERE [] IN [];", false, `[]`, nil},
```
Comment (asdine) [inline sql/query/select_test.go:92]: @goku321 Could you add a link to the issue next to this test?

**Review by asdine — COMMENTED** [review]
Thanks @goku321!

File: sql/query/select_test.go (hunk 1 above)
Comment (goku321) [reply sql/query/select_test.go:92]: Sure

**Review by asdine — APPROVED** [review]
Thanks!

**Pull request merged**